piper/buttondialog.py

```diff
@@ -294,8 +296,13 @@ def _on_macro_updated(self, macro, pspec):
 
     def _on_macro_set(self, macro):
         # A macro has been set; update accordingly.
-        self._action_type = RatbagdButton.ActionType.MACRO
-        self._mapping = macro
+        if RatbagdButton.ActionType.MACRO in self._button.action_types:  # macro is supported
+            self._action_type = RatbagdButton.ActionType.MACRO
+            self._mapping = macro
+        else:
+            self._action_type = RatbagdButton.ActionType.KEY
+            type_, value = macro.keys[0]
+            self._mapping = value
```

Comment:
> I would like to try to reuse the macro dialog and just limit it to single key action.

That does make sense.

Well, then I think this code should use the `key` action type opportunistically, falling back to macro and giving an error message if needed.

Pseudo-code:
```
if SupportsKeyActionType() and macro.keys.len() == 1:
    // SetKeyActionType()
    return
SetMacroActionType()
```